scipy/stats/_hypotests.py

```diff
@@ -132,6 +134,227 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
+
+    See equation 1.8 in Csorgo, S. and Faraway, J. (1996) for finite samples,
+    1.2 for the asymptotic cdf.
+
+    The function is not expected to be accurate for large values of x, say
+    x > 4, when the cdf is very close to 1.
+    """
+    x = np.asarray(x)
+    if n is None:
+        y = _cdf_cvm_inf(x)
+    else:
+        # support of the test statistic is [12/n, n/3], see 1.1 in [2]
+        y = np.zeros_like(x, dtype='float')
+        sup = (1./(12*n) < x) & (x < n/3.)
+        # note: _psi1_mod does not include the term _cdf_cvm_inf(x) / 12
+        # therefore, we need to add it here
+        y[sup] = _cdf_cvm_inf(x[sup]) * (1 + 1./(12*n)) + _psi1_mod(x[sup]) / n
+        y[x >= n/3] = 1
+
+    if y.ndim == 0:
+        return y[()]
+    return y
+
+
+def cvm_test(rvs, cdf, args=()):
+    """
+    Perform the Cramer-von Mises test for goodness of fit.
+
+    This performs a test of the goodness of fit of a cumulative distribution
+    function F compared to the empirical distribution function F_n of
+    observed random variates X_1, ..., X_n that are assumed to be independent
+    and identically distributed ([1]_). The null hypothesis is that the X_i
+    have cumulative distribution F.
+
+    Parameters
+    ----------
+    rvs : array_like
+        The observed values of the random variables, it should be a 1-D array.
+    cdf : str or callable
+        If a string, it should be the name of a distribution in `scipy.stats`.
+        If a callable, that callable is used to calculate the cdf.
+        ``cdf(x, *args) -> float``, it should be a cdf.
+    args : tuple, optional
+        Distribution parameters. These are assumed to be known, see Notes.
+
+    Returns
+    -------
+    statistic : float
+        Cramer-von Mises statistic.
+    pvalue :  float
+        The p-value.
+
+    See Also
+    --------
+    kstest
+
+    Notes
+    -----
+    The p-value relies on the approximation given by equation 1.8 in [2]_.
+    It is important to keep in mind that the p-value is only accurate if
+    one tests a simple hypothesis, i.e. the parameters of the reference
+    distribution are known. If the parameters are estimated from the data
+    (composite hypothesis), the computed p-value is not reliable.
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Cramer-von_Mises_criterion
+    .. [2] Csorgo, S. and Faraway, J. (1996). The Exact and Asymptotic
+           Distribution of Cramer-von Mises Statistics. Journal of the
+           Royal Statistical Society, pp. 221-234.
+
+    Examples
+    --------
+    >>> from scipy import stats
+
+    Test the assumption of normality. In this example, one accepts
+    the assumption of a normal distribution at the significance level
+    of alpha=0.05.
+
+    >>> np.random.seed(626)
+    >>> x = stats.norm.rvs(size=500)
+    >>> w, p = stats.cvm_test(x, 'norm')
+    >>> w, p
+    (0.06342154705518796, 0.792680516270629)
```

Comment:
Maybe explain, e.g.

The *p* value 0.79 exceeds our chosen significance level, so we do not reject the null hypothesis that the observed sample is drawn from the standard normal distribution.

Reply:
done